src/js/components/modals/AppModalComponent.jsx

```diff
@@ -112,7 +112,8 @@ var AppModalComponent = React.createClass({
     // All status below 300 are actually not an error
     if (status < 300) {
       this.onCreateApp();
-    } else if (status === 409) {
+    } else if (status === 409 && data.deployments != null) {
```

Comment:
As I read the Scala, the presence of `deployments` indicates a deployment error so it's actually more logical not to check for zero-length. I can't see a situation in which an empty array would be sent so it wouldn't do any harm to add this check though.